app/sessionStore.js

```diff
@@ -868,8 +869,25 @@ const fingerprintingProtectionMigration = (immutableData) => {
   return immutableData
 }
 
+const protocolHandlerMigration = (immutableData) => {
+  // Fix https://github.com/brave/browser-laptop/issues/12797
+  if (!immutableData.get('defaultProtocolMigration')) {
+    if (platformUtil.isWindows() &&
+      immutableData.getIn(['settings', settings.IS_DEFAULT_BROWSER])) {
+      // Update the protocol handler to be safe
+      defaultProtocols.forEach((protocol) => {
+        app.removeAsDefaultProtocolClient(protocol)
+        app.setAsDefaultProtocolClient(protocol, undefined, ['--'])
```

Comment:
function signature here is not right `app.setAsDefaultProtocolClient(protocol, [path, args])`
https://github.com/brave/muon/blob/80442e401dd3f78d55f78daf2f5f2d8505970296/atom/browser/browser.h#L84